tm2/pkg/crypto/keys/client/broadcast.go

```diff
@@ -133,6 +139,31 @@ func BroadcastHandler(cfg *BroadcastCfg) (*ctypes.ResultBroadcastTxCommit, error
 	return bres, nil
 }
 
+func estimateGasFee(cli client.ABCIClient, bres *ctypes.ResultBroadcastTxCommit) error {
+	gp := std.GasPrice{}
+	qres, err := cli.ABCIQuery("auth/gasprice", []byte{})
+	if err != nil {
+		return errors.Wrap(err, "query gas price")
+	}
+	err = amino.UnmarshalJSON(qres.Response.Data, &gp)
+	if err != nil {
+		return errors.Wrap(err, "unmarshaling query gas price result")
+	}
+
+	if gp.Gas == 0 {
+		return nil
+	}
+
+	fee := bres.DeliverTx.GasUsed/gp.Gas + 1
+	fee = overflow.Mulp(fee, gp.Price.Amount)
+	// 5% fee buffer to cover the suden change of gas price
+	feeBuffer := overflow.Mulp(fee, 5) / 100
+	fee = overflow.Addp(fee, feeBuffer)
+	s := fmt.Sprintf("estimated gas usage: %d, gas fee: %d%s, current gas price: %s\n", bres.DeliverTx.GasUsed, fee, gp.Price.Denom, gp.String())
```

Comment:
Is there a world in which perhaps the folks consuming the results from this query would get back structured data say as JSON in case they need to display the estimates in other contexts otherwise they have to manually parse out  the line and the format.